sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/config/ServiceBusBinderConfiguration.java

```diff
@@ -78,20 +91,71 @@ public ServiceBusChannelProvisioner serviceBusChannelProvisioner() {
      * @param bindingProperties the binding Properties
      * @param namespaceProperties the namespace Properties
      * @param messageConverter the message Converter
+     * @param customizers the client factory customizers
      * @return ServiceBusMessageChannelBinder bean the Service Bus Message Channel Binder bean
      */
     @Bean
     @ConditionalOnMissingBean
     public ServiceBusMessageChannelBinder serviceBusBinder(ServiceBusChannelProvisioner channelProvisioner,
                                                            ServiceBusExtendedBindingProperties bindingProperties,
                                                            ObjectProvider<NamespaceProperties> namespaceProperties,
-                                                           @Nullable ServiceBusMessageConverter messageConverter) {
+                                                           @Nullable ServiceBusMessageConverter messageConverter,
+                                                           ObjectProvider<ClientFactoryCustomizer> customizers) {
 
         ServiceBusMessageChannelBinder binder = new ServiceBusMessageChannelBinder(null, channelProvisioner);
         binder.setBindingProperties(bindingProperties);
         binder.setNamespaceProperties(namespaceProperties.getIfAvailable());
         binder.setMessageConverter(messageConverter);
+        binder.setClientFactoryCustomizers(customizers.orderedStream().collect(Collectors.toList()));
         return binder;
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    ClientFactoryCustomizer defaultClientFactoryCustomizer(
+        AzureTokenCredentialResolver azureTokenCredentialResolver,
+        @Qualifier(DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME) DefaultAzureCredential defaultAzureCredential) {
```

Comment:
means may have more than one beans of this Type? so what about the case of other bean of this type? ignored directly? designed this way or some other reasons?

Reply:
No, other beans can exist in the context, but the default credential bean will only use this name.

Reply:
It is by design.